gcs/innards/constraint_state.hh

```diff
@@ -0,0 +1,37 @@
+#ifndef GLASGOW_CONSTRAINT_SOLVER_GUARD_GCS_CONSTRAINT_STATE_HH
+#define GLASGOW_CONSTRAINT_SOLVER_GUARD_GCS_CONSTRAINT_STATE_HH
+#include <gcs/integer.hh>
+#include <set>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+#include <any>
+
+using namespace gcs;
+struct RegularGraph
```

Comment:
The nice thing about `std::any` is that this can now go in `regular.cc`, and be completely invisible to anyone who isn't specifically looking at your constraint.

Reply:
I guess in that case I wouldn't really need `constraint_state.hh` at all then?

Reply:
Yeah, it's probably fine just as a `using` in `state.hh`.